In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go
from IPython.display import HTML
from IPython.display import Video

In [2]:
df = pd.read_csv('test_bbb.csv')

C:\Users\range\AppData\Local\Temp\ipykernel_22208\2372166412.py:1: DtypeWarning: Columns (13,40,47,51,52) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('test_bbb.csv')


In [3]:
df.head()

Unnamed: 0   p_match  inns               bat    p_bat      team_bat  \
0           0  226372.0   2.0  Herschelle Gibbs  45224.0  South Africa   
1           1  226372.0   2.0  Herschelle Gibbs  45224.0  South Africa   
2           2  226372.0   2.0  Herschelle Gibbs  45224.0  South Africa   
3           3  226372.0   2.0  Herschelle Gibbs  45224.0  South Africa   
4           4  226372.0   2.0  Herschelle Gibbs  45224.0  South Africa   

            bowl  p_bowl  team_bowl  ball  ...  bat_out wagonX  wagonY  \
0  Glenn McGrath  6565.0  Australia   2.0  ...     True    0.0     0.0   
1  Glenn McGrath  6565.0  Australia   3.0  ...     True  114.0   182.0   
2  Glenn McGrath  6565.0  Australia   4.0  ...     True  113.0   165.0   
3  Glenn McGrath  6565.0  Australia   5.0  ...     True  120.0   195.0   
4  Glenn McGrath  6565.0  Australia   6.0  ...     True    0.0     0.0   

  wagonZone line  length                         shot  control  predscore  \
0       0.0  NaN     NaN                      NO_SHOT      1.0       -1.0   
1       6.0  NaN     NaN              FORWARD_DEFENCE      1.0       -1.0   
2       6.0  NaN     NaN        CUT_SHOT_ON_BACK_FOOT      1.0       -1.0   
3       6.0  NaN     NaN  OFF_SIDE_DRIVE_ON_BACK_FOOT      1.0       -1.0   
4       0.0  NaN     NaN                      NO_SHOT      1.0       -1.0   

   wprob  
0   -1.0  
1   -1.0  
2   -1.0  
3   -1.0  
4   -1.0  

[5 rows x 57 columns]

In [4]:
df.columns

Index(['Unnamed: 0', 'p_match', 'inns', 'bat', 'p_bat', 'team_bat', 'bowl',
       'p_bowl', 'team_bowl', 'ball', 'ball_id', 'outcome', 'score', 'out',
       'dismissal', 'p_out', 'over', 'noball', 'wide', 'byes', 'legbyes',
       'cur_bat_runs', 'cur_bat_bf', 'cur_bowl_ovr', 'cur_bowl_wkts',
       'cur_bowl_runs', 'inns_runs', 'inns_wkts', 'inns_balls', 'trail_by',
       'lead_by', 'day', 'session', 'target', 'date', 'year', 'ground',
       'country', 'winner', 'toss', 'competition', 'bat_hand', 'bowl_style',
       'bowl_kind', 'batruns', 'ballfaced', 'bowlruns', 'bat_out', 'wagonX',
       'wagonY', 'wagonZone', 'line', 'length', 'shot', 'control', 'predscore',
       'wprob'],
      dtype='object')

In [5]:
df['bat'].unique()

array(['Herschelle Gibbs', 'AB de Villiers', 'Shaun Pollock',
       'Ashwell Prince', 'Andre Nel', 'Makhaya Ntini', nan,
       'Gautam Gambhir', 'Virender Sehwag', 'VVS Laxman',
       'Sachin Tendulkar', 'Yuvraj Singh', 'MS Dhoni', 'Irfan Pathan',
       'Ajit Agarkar', 'Anil Kumble', 'Harbhajan Singh', 'Mohammad Kaif',
       'Marvan Atapattu', 'Kumar Sangakkara', 'Upul Tharanga',
       'Mahela Jayawardene', 'Thilan Samaraweera', 'Tillakaratne Dilshan',
       'Malinga Bandara', 'Lasith Malinga', 'Farveez Maharoof',
       'Jehan Mubarak', 'Muthiah Muralidaran', 'Ricky Ponting',
       'Justin Langer', 'Matthew Hayden', 'Brad Hodge', 'Michael Hussey',
       'Andrew Symonds', 'Adam Gilchrist', 'Shane Warne', 'Brett Lee',
       'Nathan Bracken', 'Glenn McGrath', 'Graeme Smith', 'Justin Kemp',
       'Mark Boucher', 'Jacques Rudolph', 'Charl Langeveldt',
       'Rahul Dravid', 'Sourav Ganguly', 'Avishka Gunawardene',
       'Dilhara Fernando', 'Chaminda Vaas', 'Mohammad Yousuf',
  

In [6]:
Steve_smith_test = df[df['bat']=='Steven Smith']

In [7]:
Steve_smith_test.head()

Unnamed: 0   p_match  inns           bat     p_bat   team_bat  \
404903      404903  428752.0   1.0  Steven Smith  267192.0  Australia   
404904      404904  428752.0   1.0  Steven Smith  267192.0  Australia   
404905      404905  428752.0   1.0  Steven Smith  267192.0  Australia   
404906      404906  428752.0   1.0  Steven Smith  267192.0  Australia   
404907      404907  428752.0   1.0  Steven Smith  267192.0  Australia   

                  bowl  p_bowl team_bowl  ball  ...  bat_out wagonX  wagonY  \
404903  James Anderson  8608.0   England   1.0  ...     True    0.0     0.0   
404904  James Anderson  8608.0   England   2.0  ...     True    0.0     0.0   
404905  James Anderson  8608.0   England   3.0  ...     True  171.0   107.0   
404906  James Anderson  8608.0   England   3.0  ...     True    0.0     0.0   
404907  James Anderson  8608.0   England   4.0  ...     True  217.0   240.0   

       wagonZone line  length                         shot  control  \
404903       0.0  NaN     NaN              FORWARD_DEFENCE      0.0   
404904       0.0  NaN     NaN                      NO_SHOT      1.0   
404905       8.0  NaN     NaN              FORWARD_DEFENCE      0.0   
404906       0.0  NaN     NaN              FORWARD_DEFENCE      0.0   
404907       4.0  NaN     NaN  ON_SIDE_DRIVE_ON_FRONT_FOOT      1.0   

        predscore  wprob  
404903       -1.0   -1.0  
404904       -1.0   -1.0  
404905       -1.0   -1.0  
404906       -1.0   -1.0  
404907       -1.0   -1.0  

[5 rows x 57 columns]

In [8]:
Steve_smith_test['bowl_style'] = Steve_smith_test['bowl_style'].replace({'RFM': 'RMF', 'RM': 'RMF'})

C:\Users\range\AppData\Local\Temp\ipykernel_22208\4027765873.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Steve_smith_test['bowl_style'] = Steve_smith_test['bowl_style'].replace({'RFM': 'RMF', 'RM': 'RMF'})


In [9]:
shot_area_mapping = {
    0: 'Defended',
    1: 'Towards Third Man',
    2: 'Towards Fine Leg',
    3: 'Towards Square Leg',
    4: 'Towards Mid Wicket',
    5: 'Towards Long On',
    6: 'Towards Long Off',
    7: 'Towards Covers',
    8: 'Towards Point'
}

Steve_smith_test['Shot Area'] = Steve_smith_test['wagonZone'].map(shot_area_mapping)

C:\Users\range\AppData\Local\Temp\ipykernel_22208\1677031861.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Steve_smith_test['Shot Area'] = Steve_smith_test['wagonZone'].map(shot_area_mapping)


#### Lets see in which areas does Steven Smith scores most runs in test cricket 

In [10]:
runs_by_area = Steve_smith_test.groupby('Shot Area')['batruns'].sum().reset_index()

total_runs = runs_by_area['batruns'].sum()

runs_by_area['Percentage'] = round(((runs_by_area['batruns'] / total_runs) * 100),2)

runs_by_area = runs_by_area.sort_values(by='batruns', ascending=False)
runs_by_area

Shot Area  batruns  Percentage
7  Towards Square Leg   1933.0       22.18
3    Towards Long Off   1766.0       20.26
2    Towards Fine Leg   1297.0       14.88
5  Towards Mid Wicket    968.0       11.11
8   Towards Third Man    735.0        8.43
1      Towards Covers    730.0        8.38
4     Towards Long On    700.0        8.03
6       Towards Point    576.0        6.61
0            Defended     10.0        0.11

#### Lets see how does steven smith gets dismissed mostly in test cricket

In [11]:
Steve_smith_dismissals = Steve_smith_test[Steve_smith_test['bat_out'] == True]

dismissal_counts = Steve_smith_dismissals.groupby('dismissal').size().reset_index(name='Count')

total_dismissals = dismissal_counts['Count'].sum()
dismissal_counts['Percentage'] = round(((dismissal_counts['Count'] / total_dismissals) * 100),2)

dismissal_counts = dismissal_counts.sort_values(by='Count', ascending=False)

print(dismissal_counts)

           dismissal  Count  Percentage
1             caught     92       57.86
0             bowled     29       18.24
2  leg before wicket     29       18.24
3            run out      5        3.14
4            stumped      4        2.52


#### The table below provides a detailed breakdown of the types of deliveries that have successfully led to the dismissal of Steven Smith in Test cricket

In [12]:
dismissal_breakdown = Steve_smith_dismissals.groupby(['line', 'length', 'bowl_style', 'dismissal']).size().reset_index(name='Count')

dismissal_breakdown['Total Dismissals'] = dismissal_breakdown.groupby(['line', 'length', 'bowl_style'])['Count'].transform('sum')

dismissal_breakdown = dismissal_breakdown.sort_values(by='Count', ascending=False)

dismissal_table = dismissal_breakdown[['line', 'length', 'bowl_style', 'dismissal', 'Count']]

pd.set_option('display.max_rows', None); pd.set_option('display.max_columns', None); dismissal_table


line                  length bowl_style  \
45       OUTSIDE_OFFSTUMP             GOOD_LENGTH        RMF   
46       OUTSIDE_OFFSTUMP             GOOD_LENGTH        RMF   
58       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        RMF   
55       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         RF   
47       OUTSIDE_OFFSTUMP             GOOD_LENGTH        SLA   
32       OUTSIDE_OFFSTUMP                    FULL        RMF   
57       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        RMF   
43       OUTSIDE_OFFSTUMP             GOOD_LENGTH         RF   
38       OUTSIDE_OFFSTUMP             GOOD_LENGTH         OB   
5                DOWN_LEG                   SHORT        LMF   
26          ON_THE_STUMPS             GOOD_LENGTH        SLA   
25          ON_THE_STUMPS             GOOD_LENGTH        SLA   
19          ON_THE_STUMPS             GOOD_LENGTH         OB   
39       OUTSIDE_OFFSTUMP             GOOD_LENGTH         OB   
27          ON_THE_STUMPS                   SHORT         RF   
21          ON_THE_STUMPS             GOOD_LENGTH         RF   
44       OUTSIDE_OFFSTUMP             GOOD_LENGTH        RMF   
10          ON_THE_STUMPS                    FULL         OB   
70  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH        SLA   
53       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        LBG   
74  WIDE_OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         RF   
73  WIDE_OUTSIDE_OFFSTUMP                   SHORT        RMF   
72  WIDE_OUTSIDE_OFFSTUMP                   SHORT        RMF   
71  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH        SLA   
48       OUTSIDE_OFFSTUMP             GOOD_LENGTH        SLA   
49       OUTSIDE_OFFSTUMP             GOOD_LENGTH        SLA   
50       OUTSIDE_OFFSTUMP                   SHORT        RMF   
51       OUTSIDE_OFFSTUMP                   SHORT        RMF   
52       OUTSIDE_OFFSTUMP                   SHORT        SLA   
62  WIDE_OUTSIDE_OFFSTUMP                    FULL        LFM   
61       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        SLA   
69  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH        RMF   
68  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH         RF   
67  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH         OB   
56       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         RF   
66  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH        LFM   
65  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH         LF   
64  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH        LBG   
59       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        RMF   
63  WIDE_OUTSIDE_OFFSTUMP                    FULL         RF   
60       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        SLA   
54       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         LF   
0                DOWN_LEG                    FULL         OB   
42       OUTSIDE_OFFSTUMP             GOOD_LENGTH         RF   
18          ON_THE_STUMPS             GOOD_LENGTH        LFM   
2                DOWN_LEG                    FULL        SLA   
3                DOWN_LEG             GOOD_LENGTH        LBG   
4                DOWN_LEG             GOOD_LENGTH         OB   
6                DOWN_LEG  SHORT_OF_A_GOOD_LENGTH        LMF   
7           ON_THE_STUMPS                    FULL        LBG   
8           ON_THE_STUMPS                    FULL        LWS   
9           ON_THE_STUMPS                    FULL         OB   
11          ON_THE_STUMPS                    FULL         RF   
12          ON_THE_STUMPS                    FULL        SLA   
13          ON_THE_STUMPS                    FULL        SLA   
14          ON_THE_STUMPS                    FULL        SLA   
15          ON_THE_STUMPS             GOOD_LENGTH        LBG   
16          ON_THE_STUMPS             GOOD_LENGTH        LBG   
17          ON_THE_STUMPS             GOOD_LENGTH        LFM   
20          ON_THE_STUMPS             GOOD_LENGTH         RF   
41       OUTSIDE_OFFSTUMP             GOOD_LENGTH         RF   
22          ON_THE_STUMPS             GOOD_LENGTH    

### Key Insights:
- **Line and Length**: Smith has been most vulnerable to deliveries outside off-stump on a good length. This line and length combination has led to numerous dismissals, either through catches or leg before wicket (LBW).
- **Bowling Styles**: Right-arm medium-fast (RMF) bowlers have had considerable success against Smith, particularly with good-length deliveries outside off-stump. Spin bowlers, particularly left-arm orthodox (SLA), have also had success against smith.
- **Type of Dismissal**: Caught dismissals dominate the dataset, with leg before wicket and bowled also being frequent modes of getting Smith out. Notably, a few dismissals have occurred from deliveries on the stumps or down the leg side.

#### The table below provides a detailed breakdown of the Runs Of Steven Smith against the types of deliveries in Test cricket

In [13]:
total_runs = Steve_smith_test['batruns'].sum()

runs_by_combination = Steve_smith_test.groupby(['line', 'length', 'bowl_style'])['batruns'].sum().reset_index()

runs_by_combination['Percentage'] = (runs_by_combination['batruns'] / total_runs) * 100
runs_by_combination['Percentage'] = runs_by_combination['Percentage'].round(2)

runs_by_combination = runs_by_combination.sort_values(by='batruns', ascending=False)

runs_by_combination.rename(columns={'batruns': 'Total Runs'}, inplace=True)

runs_by_combination

line                  length bowl_style  Total Runs  \
143       OUTSIDE_OFFSTUMP             GOOD_LENGTH        RMF       518.0   
121       OUTSIDE_OFFSTUMP                    FULL        RMF       464.0   
165       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        RMF       390.0   
139       OUTSIDE_OFFSTUMP             GOOD_LENGTH         OB       352.0   
82           ON_THE_STUMPS             GOOD_LENGTH        SLA       300.0   
78           ON_THE_STUMPS             GOOD_LENGTH         OB       257.0   
60           ON_THE_STUMPS                    FULL        RMF       253.0   
164       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         RF       253.0   
152       OUTSIDE_OFFSTUMP                   SHORT        RMF       251.0   
118       OUTSIDE_OFFSTUMP                    FULL         OB       225.0   
81           ON_THE_STUMPS             GOOD_LENGTH        RMF       204.0   
144       OUTSIDE_OFFSTUMP             GOOD_LENGTH        SLA       189.0   
142       OUTSIDE_OFFSTUMP             GOOD_LENGTH         RF       174.0   
209  WIDE_OUTSIDE_OFFSTUMP                    FULL        RMF       131.0   
59           ON_THE_STUMPS                    FULL         RF       123.0   
80           ON_THE_STUMPS             GOOD_LENGTH         RF       122.0   
61           ON_THE_STUMPS                    FULL        SLA       118.0   
151       OUTSIDE_OFFSTUMP                   SHORT         RF       116.0   
57           ON_THE_STUMPS                    FULL         OB       115.0   
120       OUTSIDE_OFFSTUMP                    FULL         RF       102.0   
101          ON_THE_STUMPS  SHORT_OF_A_GOOD_LENGTH        RMF        86.0   
122       OUTSIDE_OFFSTUMP                    FULL        SLA        85.0   
161       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         OB        84.0   
245  WIDE_OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        RMF        83.0   
203  WIDE_OUTSIDE_OFFSTUMP                    FULL        LFM        83.0   
234  WIDE_OUTSIDE_OFFSTUMP                   SHORT        RMF        82.0   
224  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH         OB        77.0   
207  WIDE_OUTSIDE_OFFSTUMP                    FULL         OB        76.0   
226  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH        RMF        76.0   
133       OUTSIDE_OFFSTUMP             GOOD_LENGTH        LBG        74.0   
72           ON_THE_STUMPS             GOOD_LENGTH        LBG        65.0   
166       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH        SLA        51.0   
98           ON_THE_STUMPS  SHORT_OF_A_GOOD_LENGTH         OB        46.0   
210  WIDE_OUTSIDE_OFFSTUMP                    FULL        SLA        45.0   
100          ON_THE_STUMPS  SHORT_OF_A_GOOD_LENGTH         RF        45.0   
208  WIDE_OUTSIDE_OFFSTUMP                    FULL         RF        45.0   
102          ON_THE_STUMPS  SHORT_OF_A_GOOD_LENGTH        SLA        45.0   
89           ON_THE_STUMPS                   SHORT        RMF        45.0   
25                DOWN_LEG             GOOD_LENGTH         OB        38.0   
11                DOWN_LEG                    FULL        SLA        33.0   
135       OUTSIDE_OFFSTUMP             GOOD_LENGTH        LFM        32.0   
88           ON_THE_STUMPS                   SHORT         RF        32.0   
244  WIDE_OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         RF        32.0   
126       OUTSIDE_OFFSTUMP               FULL_TOSS         OB        31.0   
202  WIDE_OUTSIDE_OFFSTUMP                    FULL         LF        30.0   
10                DOWN_LEG                    FULL        RMF        30.0   
31                DOWN_LEG                   SHORT        LFM        26.0   
148       OUTSIDE_OFFSTUMP                   SHORT        LMF        26.0   
156       OUTSIDE_OFFSTUMP  SHORT_OF_A_GOOD_LENGTH         LF        24.0   
225  WIDE_OUTSIDE_OFFSTUMP             GOOD_LENGTH         RF        24.0   
33                DOWN_LEG                   SHORT        LMF        24.0   
243  WIDE_OUTSIDE_

Smith's ability to score against right-arm medium-fast bowlers, particularly on deliveries outside off stump, is a key feature of his batting. However, deliveries on the stumps at good or full length, especially from slower bowlers, seem to constrain him

Key Observations for Dismissing Steven Smith till now:
    
* Bowling Outside Off Stump: Both seamers and spinners find significant success outside off stump, with a combination of good length and short-of-a-good-length deliveries leading to many caught dismissals. A packed slip cordon is crucial.

* Targeting the Stumps: Slow left-arm and off-spin bowlers attacking the stumps, particularly with good length or full-length deliveries, expose a vulnerability to being bowled or trapped LBW.

* Shorter Lengths: Short-of-a-good-length deliveries outside off stump, particularly from faster bowlers, tend to induce edges or false shots, leading to dismissals.

* Fuller Length Outside Off Stump: Smith can be tempted to drive at full deliveries outside off stump, especially from medium-fast bowlers, leading to edges or mistimed shots.

### So our first Bowling plan to dismiss smith will be Bowling Outside Off Stump Good Length Delivery by Right Arm Fast Bowlers 

Reason:
    
Right-arm fast bowlers targeting a good length outside off stump seems to trouble Smith consistently. His tendency to play at these deliveries which we can see in the table where he scored the most runs also in this line length too, especially against swing or seam movement, results in a higher chance of edging or missing the ball, leading to dismissals via caught behind or LBW.

Field Setup: Slip cordon, gully, and an attacking off-side field can be beneficial to maximize the chances of catching edges.

Lets see where does he scores most runs against this Outside Off Stump Good Length Delivery By Right Arm Fast Bowlers:

In [14]:
outside_offstump_data = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))
]

runs_by_shot_area = outside_offstump_data.groupby('Shot Area')['batruns'].sum().reset_index()

runs_by_shot_area.rename(columns={'batruns': 'Total Runs'}, inplace=True)

runs_by_shot_area.sort_values(by='Total Runs', ascending = False)

Shot Area  Total Runs
3    Towards Long Off       204.0
7  Towards Square Leg       102.0
6       Towards Point        97.0
2    Towards Fine Leg        85.0
1      Towards Covers        68.0
8   Towards Third Man        55.0
4     Towards Long On        42.0
5  Towards Mid Wicket        39.0
0            Defended         0.0

Lets see what shots steven smith atttempts and the dismissal counts against those shots in Outside Off Stump Good Length Delivery By Right Arm Fast Bowlers:

In [15]:
outside_offstump_data = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))
]

shot_by_dismissal = outside_offstump_data.groupby('shot').agg(total_shots = ('shot', 'size'),dismissals=('dismissal', lambda x: x.notna().sum()) 
).reset_index()

shot_by_dismissal.sort_values(by='total_shots', ascending = False)

shot  total_shots  dismissals
3         DEFENDED         1138          11
5       LEFT_ALONE          457           2
4            FLICK          222           3
0      COVER_DRIVE          140           2
7         ON_DRIVE           81           1
10            PUSH           79           1
13         STEERED           56           0
1         CUT_SHOT           26           0
6       LEG_GLANCE           23           0
12    SQUARE_DRIVE           20           0
11       SLOG_SHOT            6           0
2              DAB            3           0
14  STRAIGHT_DRIVE            2           0
8      PADDLE_AWAY            1           0
9             PULL            1           0

In [29]:
outside_offstump_data_cd = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))&
    (Steve_smith_test['shot'] == 'COVER_DRIVE')
]

outside_offstump_data_cd['control'].value_counts()

control
1.0    121
0.0     19
Name: count, dtype: int64

So another thing we notice here is :

* Steven Smith Has High defensive play in this outside off stump good length deliveries by right arm fast bowlers
(Defended: 1,138 shots, 11 dismissals.)
* This is by far the most frequent response from Steven Smith when faced with deliveries in this area. The defensive shots result in the highest number of dismissals (11), making it a key strategy. His tendency to defend exposes him to potential dismissals, particularly if there’s some movement off the pitch or in the air.

Lets see his kind of dismissals while defending the ball in this outside off stump good length deliveries by right arm fast bowlers: 

In [16]:
outside_offstump_data_defend = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))&
    (Steve_smith_test['shot'] == 'DEFENDED')&
    (Steve_smith_test['out']== True)
]

dismissal_counts = outside_offstump_data_defend.groupby('dismissal').size().reset_index(name='Count')

dismissal_counts.sort_values(by='Count', ascending = False)

dismissal  Count
2  leg before wicket      5
0             bowled      3
1             caught      3

In [30]:
outside_offstump_data_defend_ = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))&
    (Steve_smith_test['shot'] == 'DEFENDED')
]

outside_offstump_data_defend_['control'].value_counts()

control
 1.0    962
 0.0    173
-1.0      3
Name: count, dtype: int64

In [17]:
outside_offstump_data_defend[['dismissal', 'control', 'bowl', 'team_bowl', 'date']]

dismissal  control            bowl     team_bowl        date
865232              caught      0.0     Steven Finn       England  2015-07-29
917259              caught      0.0     Kyle Abbott  South Africa  2016-11-24
1002658  leg before wicket      0.0    Chris Woakes       England  2017-12-02
1002893             bowled      0.0   Craig Overton       England  2017-12-02
1285901             bowled      0.0  James Anderson       England  2021-12-26
1286657  leg before wicket      0.0  James Anderson       England  2021-12-16
1376480             bowled      0.0   Anrich Nortje  South Africa  2022-12-17
1462861  leg before wicket      0.0       Mark Wood       England  2023-07-19
1472787  leg before wicket      0.0      Ben Stokes       England  2023-06-16
1515300  leg before wicket      0.0      Matt Henry   New Zealand  2024-03-08
1517684             caught      0.0      Matt Henry   New Zealand  2024-02-29

In [18]:
outside_offstump_data_defend

Unnamed: 0    p_match  inns           bat     p_bat   team_bat  \
865232       865232   743967.0   1.0  Steven Smith  267192.0  Australia   
917259       917259  1000855.0   4.0  Steven Smith  267192.0  Australia   
1002658     1002658  1072306.0   3.0  Steven Smith  267192.0  Australia   
1002893     1002893  1072306.0   1.0  Steven Smith  267192.0  Australia   
1285901     1285901  1263464.0   2.0  Steven Smith  267192.0  Australia   
1286657     1286657  1263463.0   1.0  Steven Smith  267192.0  Australia   
1376480     1376480  1317492.0   2.0  Steven Smith  267192.0  Australia   
1462861     1462861  1336046.0   1.0  Steven Smith  267192.0  Australia   
1472787     1472787  1336043.0   2.0  Steven Smith  267192.0  Australia   
1515300     1515300  1388227.0   4.0  Steven Smith  267192.0  Australia   
1517684     1517684  1388226.0   1.0  Steven Smith  267192.0  Australia   

                   bowl    p_bowl     team_bowl  ball  ball_id outcome  score  \
865232      Steven Finn  210283.0       England   6.0     7.06     out    0.0   
917259      Kyle Abbott  297583.0  South Africa   6.0    39.06     out    0.0   
1002658    Chris Woakes  247235.0       England   2.0    23.02     out    0.0   
1002893   Craig Overton  464626.0       England   1.0    62.01     out    0.0   
1285901  James Anderson    8608.0       England   1.0    35.01     out    0.0   
1286657  James Anderson    8608.0       England   6.0   138.06     out    0.0   
1376480   Anrich Nortje  481979.0  South Africa   2.0    32.02     out    0.0   
1462861       Mark Wood  351588.0       England   2.0    29.02     out    0.0   
1472787      Ben Stokes  311158.0       England   7.0    26.07     out    0.0   
1515300      Matt Henry  506612.0   New Zealand   1.0     7.01     out    0.0   
1517684      Matt Henry  506612.0   New Zealand   1.0    24.01     out    0.0   

          out          dismissal     p_out   over  noball  wide  byes  \
865232   True             caught  267192.0    8.0     0.0   0.0   0.0   
917259   True             caught  267192.0   40.0     0.0   0.0   0.0   
1002658  True  leg before wicket  267192.0   24.0     0.0   0.0   0.0   
1002893  True             bowled  267192.0   63.0     0.0   0.0   0.0   
1285901  True             bowled  267192.0   36.0     0.0   0.0   0.0   
1286657  True  leg before wicket  267192.0  139.0     0.0   0.0   0.0   
1376480  True             bowled  267192.0   33.0     0.0   0.0   0.0   
1462861  True  leg before wicket  267192.0   30.0     0.0   0.0   0.0   
1472787  True  leg before wicket  267192.0   27.0     0.0   0.0   0.0   
1515300  True  leg before wicket  267192.0    8.0     0.0   0.0   0.0   
1517684  True             caught  267192.0   25.0     0.0   0.0   0.0   

         legbyes  cur_bat_runs  cur_bat_bf  cur_bowl_ovr  cur_bowl_wkts  \
865232       0.0           7.0        18.0           1.0            1.0   
917259       0.0          40.0        52.0          10.0            1.0   
1002658      0.0           6.0        12.0           5.2            2.0   
1002893      0.0          40.0        90.0          10.1            1.0   
1285901      0.0          16.0        31.0          13.1            2.0   
1286657      0.0          93.0       201.0          26.0            1.0   
1376480      0.0          36.0        68.0           7.2            2.0   
1462861      0.0          41.0        52.0           6.2            1.0   
1472787      0.0          16.0        59.0           2.0            1.0   
1515300      0.0           9.0        25.0           3.1            1.0   
1517684      0.0          31.0        71.0           7.1            1.0   

         cur_bowl_runs  inns_runs  inns_wkts  inns_balls  trail_by  lead_by  \
865232             4.0       18.0        2.0        48.0       0.0      0.0   
917259            26.0      125.0        3.0       240.0       1.0      0.0   
1002658           10.0       50.0        4.0       140.0       0.0    265.0   
1002893           32.0      161.0        

After watching and analysing all the dismissals, i got to know that outside off stump line where smith actually struggle is the 4th Stump- 5th stump line 

Dismissal Methods:    
* LBW: When the ball angles in, Smith tends to fall across his stumps or gets trapped in front. This accounts for 5 leg-before dismissals.

* Caught Behind/Slips: Outswingers on this line often get Smith chasing the ball, leading to edges to the slips or keeper, responsible for 3 caught dismissals.

* Bowled: A straight delivery that holds its line or slightly nips in has bowled him 3 times.

* Smith tends to defend deliveries outside off stump heavily, with 1138 defensive shots attempted, but this also leads to 11 dismissals. This indicates vulnerability when he's pinned on the back foot or when his guard is outside off stump.
* The fact that 5 dismissals are LBW shows that when he shuffles across, he is susceptible to deliveries that angle in.
* 3 bowled dismissals reinforce that when he misjudges length or swing, his stumps are exposed.

Also one thing Steven Smith attempted 140 cover drives but only got out twice, showing his control over attacking shots in this area. Means You have to force him to play defensive shots where his control is less compared to when he looks to play attacking shots


The most effective strategy for dismissing Smith is to bowl a consistent good length around the 4th or 5th stump line with slight variations in seam or swing.

* In-swingers: Especially from RMF and RF bowlers, should be aimed at angling into Smith, targeting his pads and stumps. This exploits his tendency to shuffle and overcompensate, leading to LBWs and bowled dismissals.
* Out-swingers: Should challenge him to defend while moving his feet, forcing errors that lead to edges and caught dismissals.

#### Our Second Bowling plan to dismiss smith will be Bowling Short of a Good Length, Outside Off Stump, Deliveries by RMF & RF Bowlers

Reason:

* Smith has been dismissed several times by deliveries that are Outside Off Stump and short of a good length (15 Times), especially from right-arm fast bowlers (RMF/RF). This length forces him to make decisions quickly—whether to go back and defend or attack.
* Smith's backfoot play is solid, but deliveries around chest height or just short of his comfort zone (4th and 5th stump) can force him into mistakes, especially when the ball is moving

Lets see where does he scores most runs against this Outside Off Stump Short Of A Good Length Delivery By Right Arm Fast Bowlers:

In [19]:
outside_offstump_data_1 = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'SHORT_OF_A_GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))
]

runs_by_shot_area_1 = outside_offstump_data.groupby('Shot Area')['batruns'].sum().reset_index()

runs_by_shot_area_1.rename(columns={'batruns': 'Total Runs'}, inplace=True)

runs_by_shot_area_1.sort_values(by='Total Runs', ascending = False)

Shot Area  Total Runs
3    Towards Long Off       204.0
7  Towards Square Leg       102.0
6       Towards Point        97.0
2    Towards Fine Leg        85.0
1      Towards Covers        68.0
8   Towards Third Man        55.0
4     Towards Long On        42.0
5  Towards Mid Wicket        39.0
0            Defended         0.0

Strong Scoring Areas:

* Towards Long Off (204 runs): This area has been Smith’s most productive scoring zone. When faced with short of a good length deliveries, he tends to drive or punch the ball through this region, highlighting his tendency to play on the up or use his strong backfoot game to hit straight.
* Towards Square Leg (102 runs) and Towards Point (97 runs): These two regions are also key scoring areas for Smith. The runs towards square leg could indicate that Smith occasionally turns these deliveries off his hips or wrists into the leg side. The runs towards point suggest he often cuts or steers the ball into this area.

The relatively low scoring in Mid Wicket areas suggests that Smith is not consistently targeting these zones when facing deliveries on this line and length. This implies he is not frequently getting across the ball to work it into the leg side, keeping his scoring patterns more off-side dominant

Lets see what shots steven smith atttempts and the dismissal counts against those shots in Outside Off Stump Short Of A Good Length Delivery By Right Arm Fast Bowlers:

In [20]:
outside_offstump_data_1 = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'SHORT_OF_A_GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))
]

shot_by_dismissal_1 = outside_offstump_data_1.groupby('shot').agg(total_shots = ('shot', 'size'),dismissals=('dismissal', lambda x: x.notna().sum()) 
).reset_index()

shot_by_dismissal_1.sort_values(by='total_shots', ascending= False)

shot  total_shots  dismissals
3         DEFENDED          611          11
6       LEFT_ALONE          471           0
4            FLICK          180           1
11            PUSH           74           1
1         CUT_SHOT           70           2
10            PULL           70           0
0      COVER_DRIVE           54           1
14         STEERED           48           0
13    SQUARE_DRIVE           42           0
8         ON_DRIVE           13           0
7       LEG_GLANCE           12           0
2              DAB            6           0
12       SLOG_SHOT            3           1
5             HOOK            2           0
15  STRAIGHT_DRIVE            2           0
9      PADDLE_AWAY            1           0

Defensive shots form a large portion of Smith's approach against this length, but they also account for the highest number of dismissals (11). This shows that even though Smith often looks to defend, the nature of the short of a good length delivery allows bowlers to beat his defense, either through seam movement or bounce, leading to bowled or caught dismissals.
Bowlers should continue attacking this length with the expectation that Smith’s defensive play is vulnerable, especially when there is lateral movement.

Lets see more about how smith gets dismissed while defending the outside off stump short of a good length ball :

In [21]:
outside_offstump_data_defend_1 = Steve_smith_test[
    (Steve_smith_test['line'] == 'OUTSIDE_OFFSTUMP') &
    (Steve_smith_test['length'] == 'SHORT_OF_A_GOOD_LENGTH') &
    ((Steve_smith_test['bowl_style'] == 'RMF') | (Steve_smith_test['bowl_style'] == 'RF'))&
    (Steve_smith_test['shot'] == 'DEFENDED')&
    (Steve_smith_test['out']== True)
]

dismissal_counts_1 = outside_offstump_data_defend_1.groupby('dismissal').size().reset_index(name='Count')

dismissal_counts_1.sort_values(by='Count', ascending = False)

dismissal  Count
1             caught      8
0             bowled      2
2  leg before wicket      1

In [23]:
outside_offstump_data_defend_1[['dismissal', 'control', 'team_bowl','bowl','date','year']]

dismissal  control     team_bowl             bowl  \
745600              bowled      0.0         India    Ishant Sharma   
1071572  leg before wicket      0.0         India      Umesh Yadav   
1157076             caught      0.0  South Africa     Morne Morkel   
1290282             caught      0.0       England        Mark Wood   
1358213             caught      0.0         India   Mohammed Siraj   
1438110             caught      0.0       England   Ollie Robinson   
1450519             caught      0.0      Pakistan      Aamer Jamal   
1452245             caught      0.0      Pakistan  Khurram Shahzad   
1458339             caught      0.0       England     Chris Woakes   
1475387             bowled      0.0         India   Shardul Thakur   
1536808             caught      0.0   West Indies    Shamar Joseph   

               date    year  
745600   2014-12-17  2014.0  
1071572  2017-03-04  2017.0  
1157076  2018-03-22  2018.0  
1290282  2021-12-08  2021.0  
1358213  2021-01-15  2021.0  
1438110  2022-01-14  2022.0  
1450519  2023-12-26  2023.0  
1452245  2023-12-14  2023.0  
1458339  2023-07-27  2023.0  
1475387  2023-06-07  2023.0  
1536808  2024-01-17  2024.0

Observation after watching the above mentioned dismissals after 2020:

* Ball ID 1536808 – Delivered around the 5th-6th stump line, bouncing to stump height with a hint of outswing.
* Ball ID 1475387 – Pitched on the 5th-6th stump line, rising to stump height, and held its line after pitching.
* Ball ID 1458339 – Targeted the 5th stump line, bouncing to stump height with slight outswing.
* Ball ID 1452245 – Landed around the 5th-6th stump line, bouncing two inches above the stumps with noticeable outswing.
* Ball ID 1450519 – Delivered on the 4th stump line, rising to stump height with a touch of outswing.
* Ball ID 1438110 – Bowled on the 4th stump line, bouncing up to stomach height and holding its line.
* Ball ID 1358213 – Targeted the 4th stump line, rising to chest height, and jagged inwards after pitching.
* Ball ID 1290282 – targeted on the 5th stump line, bouncing below chest height with slight inward movement after pitching.

When the ball is pitched just outside off stump, slightly short of a good length around the 5th-6th stump line, and keeps low—bouncing only up to stump height with some outswing or slight outswing—there’s a high probability of inducing an outside edge from Steven Smith if he attempts a defensive shot. Similarly, when the ball is delivered on a similar line and length, around the 4th-5th stump line, but bounces up to chest height with some inswing, Smith is more likely to produce an inside edge.

So our Plan B is Divided into two parts :

1. Short of a Good Length, 4th-6th Stump Line, Bouncing to Stump Height (Stump Height Target)

 Objective: Induce Edges with Outswing

- Line: Focus on the 5th or 6th stump line, just outside off, encouraging Smith to defend the ball.
- Length: The ball should rise no higher than the stumps. This keeps Smith in his crease, making him vulnerable to swing.

 Variation: 
- Slight outswing is key. Balls that swing away from Smith at this height are difficult to leave, especially on the back foot.

 Dismissal Focus:
- Caught behind or in slips: Smith’s defensive technique can lead to an edge when the ball is angling away from him. The dismissals in 2023 and 2024 show that low-bouncing deliveries with outswing have consistently gotten Smith caught at point or in the slips.
- LBW: If the ball comes in slightly, Smith is prone to being trapped LBW. Inward movement can jam him as he’s stuck in his crease.


2. Short of a Good Length, 4th-5th Stump Line, Bouncing to Chest Height (Chest Height Target)

Objective: Use Extra Bounce to Pin Smith on the Back Foot

- Line: Target the 4th or 5th stump line. This is more aggressive, making Smith uncomfortable with the bounce.
- Length: The ball should rise to chest height. The steeper bounce forces Smith to play defensively, which is more likely to produce an inside edge.

Variation:
- Seam movement, especially if the ball jags in after pitching, creates significant discomfort for Smith. Chest-height deliveries challenge his balance and decision-making.

Dismissal Focus:
- Caught: Smith’s dismissals to Mark Wood and Mohammed Siraj were a result of short-of-good-length deliveries that climbed to chest height, forcing mistimed defensive shots that resulted in catches.
- Bowled: A ball that jagged back from the 4th-5th stump line has also led to Smith being bowled, as seen in 2023 and 2021.
